storage_service/locations/models/dspace_rest.py

```diff
@@ -0,0 +1,379 @@
+                                              verbose_name=_l("Default ArchivesSpace repository"),
+                                              help_text=_l('Number of the default ArchivesSpace repository'))
+
+    archivesspace_archival_object = models.CharField(max_length=64,
+                                              verbose_name=_l("Default ArchivesSpace archival object"),
+                                              help_text=_l('Number of the default ArchivesSpace archival object'))
+
+    class Meta:
+        verbose_name = _l("DSpace via REST API")
+        app_label = 'locations'
+
+    ALLOWED_LOCATION_PURPOSE = [
+        Location.AIP_STORAGE,
+        Location.DIP_STORAGE,
+    ]
+
+    def __str__(self):
+        return 'space: {s.space_id}; rest_url: {s.rest_url}; user: {s.user}'.format(s=self)
+
+    def browse(self, path):
+        raise NotImplementedError(_('Dspace does not implement browse'))
+
+    def delete_path(self, delete_path):
+        raise NotImplementedError(_('DSpace does not implement deletion'))
+
+    def move_to_storage_service(self, src_path, dest_path, dest_space):
+        """ Moves src_path to dest_space.staging_path/dest_path. """
+        raise NotImplementedError(_('DSpace does not implement fetching packages'))
+
+    def _getdmdids(self, parentobj, dmdids=None):
+        if parentobj is not None:
+            if dmdids is None:
+                dmdids = {}
+            for obj in parentobj:
+                if obj.get('DMDID'):
+                    dmdids[obj.get('DMDID')] = obj.get('LABEL')
+                self._getdmdids(obj, dmdids)
+            return dmdids
+
+    def _get_metadata(self, input_path, aip_uuid, package_type):
+        """Get metadata for DSpace from METS file."""
+        output_dir = os.path.dirname(input_path) + '/'
+        dirname = os.path.splitext(os.path.basename(input_path))[0]
+        mets_path = ''
+
+        if package_type == 'AIP':
+            # Extract METS file
+            # TODO Should output dir be a temp dir?
+            relative_mets_path = os.path.join(dirname, 'data', 'METS.' + aip_uuid + '.xml')
+            mets_path = os.path.join(output_dir, relative_mets_path)
+            command = ['unar', '-force-overwrite', '-o', output_dir, input_path, relative_mets_path]
+            try:
+                p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                output, err = p.communicate()
+                LOGGER.info('Output: %s', output)
+            except subprocess.CalledProcessError:
+                LOGGER.error('Could not extract %s from %s', mets_path, input_path, exc_info=True)
+                return {}
+        elif package_type == 'DIP':
+            for dip_file in os.listdir(input_path):
+                if dip_file.startswith('METS') and dip_file.endswith(".xml"):
+                    LOGGER.info("Dip file: %s", dip_file)
+                    mets_path = os.path.join(input_path, dip_file)
```

Comment:
Yes. Of course.

I had problems accessing the AIP UUID during the DIP function call. I might be misunderstanding something but this taken from you on slack June 9th:

"So via the SS Django ORM you should be able to get from a DIP package to its source AIP package. (I am a bit confused about how this actually happens because looking at the log of the "Store DIP" micro-service, I see that the request body object contains `'related_package_uuid': None,`. The log of my "Store AIP" micro-service, on the other hand, references the DIP in its request body: `'related_package_uuid': u'010f4a96-b25e-4688-b6f5-5d186d957c79',`)."

Can you come up with a way to access it?